batch-index/src/main/java/no/unit/nva/search/BatchIndexer.java

```diff
@@ -71,9 +72,11 @@ private ListingResult fetchNextPageOfFilenames() {
     private List<String> insertPublishedPublicationsToIndex(UnixPath file) {
         logger.info("Indexing file:" + file.toString());
         Stream<JsonNode> fileContents = fetchFileContents(file);
-        Stream<Publication> publishedPublications = keepOnlyPublishedPublications(fileContents);
-        Stream<Try<SortableIdentifier>> indexActions = insertToIndex(publishedPublications);
-        List<String> failures = collectFailures(indexActions).collect(Collectors.toList());
+        List<IndexDocument> publishedPublications = keepOnlyPublishedPublications(fileContents)
+            .map(IndexDocument::fromPublication)
+            .collect(Collectors.toList());
+        List<BulkResponse> result = elasticSearchRestClient.batchInsert(publishedPublications);
+        List<String> failures = collectFailures(result).collect(Collectors.toList());
```

Comment:
I think it's a bit weird that the entire class simply returns a list of errors. For me, the IndexingResult class acts more like an interface for this class.